core/src/main/java/org/apache/iceberg/FastAppend.java

```diff
@@ -121,7 +149,8 @@ protected void cleanUncommitted(Set<ManifestFile> committed) {
       deleteFile(newManifest.path());
     }
 
-    for (ManifestFile manifest : appendManifests) {
+    // rewritten manifests are always owned by the table
```

Comment:
Similar nit: would be nice to note why the `appendManifests` are not cleaned up.

Reply:
Done